### ISSUE TITLE
Controls with top_level enabled don't response mouse event

### ISSUE THREAD

**Issue by gongpha** [godotengine/godot#44334]
**Godot version:**

Master build [`0762484`](https://github.com/godotengine/godot/commit/0762484473c1330c5d524d5cc3644d5e2b2e557c)

**OS/device including version:**

Windows 10 64-bits ( Vulkan )

**Issue description:**

From #44318, Grabber in `EditorSpinSplider` enabled `top_level` and doesn't respond when the mouse was entered. I tested more on other controls and it occurred the same as a grabber

**Steps to reproduce:**

1. Create any `Control` node.
2. Enable "Top Level" property
3. Connect `mouse_entered` or `mouse_exited` signal to a function that does something
4. Launch a game and test.

With `top_level` :
![topyes](https://user-images.githubusercontent.com/13400398/102002500-b0914600-3d2f-11eb-91f9-1d496ab81df1.gif)
WIthout `top_level` :
![topno](https://user-images.githubusercontent.com/13400398/102002502-b129dc80-3d2f-11eb-932a-90e9180582ef.gif)

**Minimal reproduction project:**

[zip](https://github.com/godotengine/godot/files/5684162/2.zip)

**Comment by gongpha** [comment]
After tracing. I found commits that made regressing this issue.
`4396e988 ~ 23a8a8c6`

<details><summary>Details</summary><p>

```
HEAD
...
b50387c8bf2737e7f100c6c32350d97301a6f2a3 not working
5f1107aa30295e686be6f41cb6d17fc2cff1e036 not working

// regression
23a8a8c667361a3f5a596da2b090889bae83f512
...
4396e98834f159da59ec790f2ff64fb65dacd9ce
// regression

a2da99f40cf2123c0906c734a2eb01e9b65a45a2 ok
cce0a27ec7dd7c11fcb37eb724b0b7e97b3b1478 ok
...
```
https://github.com/godotengine/godot/commits/master?after=01851defb5e43803bbcd6d8221a25ebbdd963119+6080&branch=master
</p></details>

**Comment by jmb462** [comment]
HEAD

c7b4dcae2f3b75ad7146e36f196893731f403144 => mouse event don't work on top_level only
543fb1c4dadd75914d595b089820aef42e691075 => no mouse **event even if not** set to top_level
4396e98834f159da59ec790f2ff64fb65dacd9ce to 499e07f0102d9969be0aaaaed281c577cacff416 didn't manage to compile... Huge refactoring on inputs and DisplayServer creation.
a2da99f40cf2123c0906c734a2eb01e9b65a45a2 working normally

Something fix mouse event between 543fb1c4dadd75914d595b089820aef42e691075 and c7b4dcae2f3b75ad7146e36f196893731f403144 without fixing it for top_level controls.

**Comment by gongpha** [comment]
This maybe regression from DisplayServer changes.

**Comment by KoBeWi** [comment]
Still valid in c630c20

**Comment by Sauermann** [comment]
I can no longer replicate this issue in current master v4.0 beta.
This was likely fixed by #59231
